app/components/sign-up.jsx

```diff
@@ -0,0 +1,351 @@
+import { createUseStyles } from 'react-jss'
+import { useAuth } from 'civil-client'
+import SvgAlertTriangle from '../svgr/alert-triangle'
+
+function SignUp(props, ref) {
+  const { className, style, onDone = () => {}, startTab = 'login', submitted = false, ...otherProps } = props
+
+  // checks if start tab requests login or sign up page
+  const [isLogIn, setIsLogIn] = useState(
+    startTab.toLowerCase().includes('up') ? false : startTab.toLowerCase().includes('in') ? true : false
+  )
+  // checks if user has pressed signed up or logged in button yet
+  const [isSubmitted, setIsSubmitted] = useState(submitted)
+  console.log('isSubmitted', isSubmitted)
+
+  const { destination, userInfo = {} } = props
+  const classes = useStyles()
+  const [state, methods] = useAuth(destination, userInfo)
+
+  // updates if the user has submitted the form and then calls the respective method
+  function handleSubmit(page) {
+    setIsSubmitted(true)
+    if (page === 'signup') {
+      methods.signup()
+    } else if (page === 'login') {
+      methods.login()
+    }
+  }
+  // if user has filled out required fields, automatically log them in
+  if (userInfo.email && userInfo.password) {
+    useEffect(() => {
+      onDone({ valid: true, value: userInfo })
+      return
+    })
+  }
+
+  // otherwise, continue showing login/sign up page
+  return (
+    <div className={cx(className, classes.SignUp)} style={style} ref={ref} {...otherProps}>
+      <div className={classes.tabs}>
+        <div className={cx(classes.tab, !isLogIn && classes.tabSelected)}>
+          <button
+            onClick={e => setIsLogIn(false)}
+            className={cx(classes.btnClick, !isLogIn && classes.btnClickSelected)}
+          >
+            Sign Up
+          </button>
+        </div>
+        <div className={cx(classes.tab, isLogIn && classes.tabSelected)}>
+          <button onClick={e => setIsLogIn(true)} className={cx(classes.btnClick, isLogIn && classes.btnClickSelected)}>
+            Log In
+          </button>
+        </div>
+      </div>
+      <div className={cx(classes.inputContainer, isLogIn ? classes.tabRightSelected : classes.tabLeftSelected)}>
+        <div className={cx(classes.inputBoxes, isLogIn && classes.disabled)}>
+          <p id="text">First Name</p>
+          <input name="first-name" className={cx(classes.input, isLogIn && classes.disabled)} />
+        </div>
+        <div className={cx(classes.inputBoxes, isLogIn && classes.disabled)}>
+          <p id="text">Last Name</p>
+          <input name="last-name" className={cx(classes.input, isLogIn && classes.disabled)} />
+        </div>
+        <div className={cx(classes.inputBoxes, !state.email && isSubmitted && classes.invalid)}>
+          <p id="text">E-mail</p>
+          <input
+            name="email"
+            className={cx(classes.input, !state.email && isSubmitted && classes.invalidInput)}
+            onChange={e => methods.onChangeEmail(e.target.value)}
```

Comment:
please make this onBlur(... instead of onChange for the email.  That way, the field won't turn red while the user is typing, but will turn read after the user moves out of that input field.

Reply:
I'm a bit confused by this. For the email and password field, the field only changes red when the user hits the submit button. The error does show up when I start typing if I don't use the onBlur(...) function, so do you want me to change the code so the field turns red after the user clicks away?

Reply:
What I see is the email input field turns red as soon as I start typing in it, and doesn't turn white until it looks like a valid email.  Because each time I type a character, onChangeEmail is getting called and it's saying the email isn't valid.  But if you wait until the user leaves the input field to call onChangeEmail, then no red.